geonetwork/src/main/java/au/org/aodn/geonetwork4/controller/Api.java

```diff
@@ -31,21 +41,120 @@
+                info.put("sourceId", metadata.getSourceInfo().getSourceId());
+
+                if(info.get(SUGGEST_LOGOS) instanceof ArrayList) {
+                    String host = setup.getSiteSetting(SiteHelper.HOST);
+                    String port = setup.getSiteSetting(SiteHelper.PORT);
+                    String protocol = setup.getSiteSetting(SiteHelper.PROTOCOL);
+                    ((ArrayList<String>) info.get(SUGGEST_LOGOS))
+                            .add(String.format("%s://%s:%s/geonetwork/images/logos/%s.png", protocol, host, port, info.get("sourceId")));
+                }
+            }
+
+            // We can also get the harvester uuid, from there we can get the harvester url
+            if(metadata.getHarvestInfo() != null) {
+
+                info.put("isHarvested", metadata.getHarvestInfo().isHarvested());
+
+                Object harvester = harvestManager.getHarvester(metadata.getHarvestInfo().getUuid());
+                if(harvester instanceof GeonetHarvester) {
+                    info.put("harvesterUri", StringUtils.removeEnd(((GeonetHarvester) harvester).getParams().host, "/"));
+                    info.put("harvesterType", "GeonetHarvester");
+                    // The geonetwork store logo under this dir with the uuid name, we provide a list of suggestion
+                    // on where to find the logo
+                    if(info.get(SUGGEST_LOGOS) instanceof ArrayList) {
+                        ((ArrayList<String>) info.get(SUGGEST_LOGOS))
+                                .add(String.format("%s/images/logos/%s.png", info.get("harvesterUri"), info.get("sourceId")));
+                    }
+                }
+                else if(harvester instanceof OaiPmhHarvester) {
+                    // Will have remote link to logo
+                    info.put("harvesterUri", StringUtils.removeEnd(((OaiPmhHarvester) harvester).getParams().url, "/"));
+                    info.put("harvesterType", "OaiPmhHarvester");
+                }
+                else if(harvester instanceof CswHarvester) {
+                    // Will have remote link to logo
+                    info.put("harvesterUri", StringUtils.removeEnd(((CswHarvester) harvester).getParams().capabUrl, "/"));
+                    info.put("harvesterType", "CswHarvester");
+                }
+                else if(harvester instanceof OgcWxSHarvester) {
+                    // Will have remote link to logo
+                    info.put("harvesterUri", StringUtils.removeEnd(((OgcWxSHarvester) harvester).getParams().url, "/"));
+                    info.put("harvesterType", "OgcWxSHarvester");
+                }
+                else if(harvester instanceof Geonet20Harvester) {
+                    // Will have remote link to logo
+                    info.put("harvesterUri", StringUtils.removeEnd(((Geonet20Harvester) harvester).getParams().host, "/"));
+                    info.put("harvesterType", "Geonet20Harvester");
+                }
```

Comment:
what will happen if the `harvester` is null or not instanceof any declared type? 

and if we can reorganise these to use some suggestions like in this thread: https://stackoverflow.com/questions/5579309/is-it-possible-to-use-the-instanceof-operator-in-a-switch-statement

Reply:
changed with null check and logging